### PR TITLE
Allow advanced blocking card to ignore Molds, Shapes and Lenses.

### PR DESCRIPTION
**Pull request by NotAPenguin0**
Original implementation by @S4mpsa, I just added support for lenses to complete the feature.

This PR makes it so the default mode of the advanced blocking card ('loose mode') ignores solidifier molds, extruder shapes and engraver lenses.

### CODE REVIEW

File: src/main/java/appeng/helpers/BlockingModeIgnoreList.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_File));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_Saw));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_Gear));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_Rotor));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_Turbine_Blade));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_Small_Gear));
+        Shapes.add(getUniqueIdentifier(ItemList.White_Dwarf_Shape_Extruder_ToolHeadDrill));
+
+        // Only use GT_OreDictUnificator if GTNH is loaded
+        if (Loader.isModLoaded("dreamcraft")) {
```
Comment (chochem) [inline src/main/java/appeng/helpers/BlockingModeIgnoreList.java:124]: you need to cover the earlier parts too

**Comment by AbdielKavash** [comment]
Here is a question: Does this work (or *should* this work) for recipes which actually use and consume a lens as a crafting ingredient? There are only a couple, but they do exist and this is something that might be not obvious for future recipe makers.

**Comment by NotAPenguin0** [comment]
This only affects what items will be ignored with blocking mode, it doesn't really care if that lens is consumed or not. Those recipes consuming a lens should still be pushed, but the lens itself will be ignored by blocking mode. I don't think this will ever break, since
- Either the lens is the only item in the recipe. In this case, blocking mode not working is fine since there would be no conflicts.
- The lens is not the only item in the recipe, in this case blocking mode will work fine because there are other items to block on.

TL;DR should work fine as far as I know.

**Comment by NotAPenguin0** [comment]
As another note, Lewis suggested blocking mode should also ignore the various blood orb tiers. I do think this is pretty easy to implement, but it would add another few `String.equals()` checks. I'm not sure how much of a hot path this is, so maybe this isn't viable performance-wise.

**Comment by AbdielKavash** [comment]
> As another note, Lewis suggested blocking mode should also ignore the various blood orb tiers. I do think this is pretty easy to implement, but it would add another few `String.equals()` checks. I'm not sure how much of a hot path this is, so maybe this isn't viable performance-wise.

Is that really a big issue? There is *one* recipe that uses a blood orb in a machine, surely the player can dedicate an assembler or a bus on the PrAss to it. There is no need to slow down *every other blocking mode check* because of this.

**Comment by NotAPenguin0** [comment]
> Is that really a big issue? There is one recipe that uses a blood orb in a machine, surely the player can dedicate an assembler or a bus on the PrAss to it. There is no need to slow down every other blocking mode check because of this.

Yeah, I don't think it is a big issue. Let's leave it at just this then.

**Review by AbdielKavash — DISMISSED** [review]
I just noticed something: this still does not ignore programmed Bio and Breakthrough circuits (`miscutils:item.BioRecipeSelector` and `miscutils:item.T3RecipeSelector` respectively). See https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/15256. Please add those as well.

**Review by Laiff — APPROVED** [review]
